core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/datetime/DateTimeFormatterUtil.java

```diff
@@ -0,0 +1,131 @@
+      .put("%j", (date) -> "DDD") // %j => DDD - (001..366)
+      .put("%k", (date) -> "H") // %k => H - (0..23)
+      .put("%l", (date) -> "h") // %l => h - (1..12)
+      .put("%p", (date) -> "a") // %p => a - AM or PM
+      .put("%M", (date) -> "LLLL") // %M => LLLL - Month name (January..December)
+      .put("%m", (date) -> "MM") // %m => MM - Month, numeric (00..12)
+      .put("%r", (date) -> "hh:mm:ss a") // %r => hh:mm:ss a - hh:mm:ss followed by AM or PM
+      .put("%S", (date) -> "ss") // %S => ss - Seconds (00..59)
+      .put("%s", (date) -> "ss") // %s => ss - Seconds (00..59)
+      .put("%T", (date) -> "HH:mm:ss") // %T => HH:mm:ss
+      .put("%W", (date) -> "EEEE") // %W => EEEE - Weekday name (Sunday..Saturday)
+      .put("%Y", (date) -> "yyyy") // %Y => yyyy - Year, numeric, 4 digits
+      .put("%y", (date) -> "yy") // %y => yy - Year, numeric, 2 digits
+      // The following are not directly supported by DateTimeFormatter.
+      .put("%D", (date) -> // %w - Day of month with English suffix
+          String.format("'%d%s'", date.getDayOfMonth(), getSuffix(date.getDayOfMonth())))
+      .put("%f", (date) -> // %f - Microseconds
+          String.format("'%d'", (date.getNano() / 1000)))
+      .put("%w", (date) -> // %w - Day of week (0 indexed)
+          String.format("'%d'", date.getDayOfWeek().getValue()))
+      .put("%U", (date) -> // %U Week where Sunday is the first day - WEEK() mode 0
+          String.format("'%d'", CalendarLookup.getWeekNumber(0, date.toLocalDate())))
+      .put("%u", (date) -> // %u Week where Monday is the first day - WEEK() mode 1
+          String.format("'%d'", CalendarLookup.getWeekNumber(1, date.toLocalDate())))
+      .put("%V", (date) -> // %V Week where Sunday is the first day - WEEK() mode 2 used with %X
+          String.format("'%d'", CalendarLookup.getWeekNumber(2, date.toLocalDate())))
+      .put("%v", (date) -> // %v Week where Monday is the first day - WEEK() mode 3 used with %x
+          String.format("'%d'", CalendarLookup.getWeekNumber(3, date.toLocalDate())))
+      .put("%X", (date) -> // %X Year for week where Sunday is the first day, 4 digits used with %V
+          String.format("'%d'", CalendarLookup.getYearNumber(2, date.toLocalDate())))
+      .put("%x", (date) -> // %x Year for week where Monday is the first day, 4 digits used with %v
+          String.format("'%d'", CalendarLookup.getYearNumber(3, date.toLocalDate())))
+      .build();
+
+  private static final String MOD = "%";
+  private static final String QUOTE_LITERAL = "'";
+
+  private DateTimeFormatterUtil() {
+  }
+
+  /**
+   * Format the date using the date format String.
+   * @param dateExpr the date ExprValue of Date/Datetime/Timestamp/String type.
+   * @param formatExpr the format ExprValue of String type.
+   * @return Date formatted using format and returned as a String.
+   */
+  static ExprValue getFormattedDate(ExprValue dateExpr, ExprValue formatExpr) {
```

Comment:
Not sure the use case. Do you try to do something like this format.replace("%\\W", () -> replace).
If so, i recommended to consider the https://docs.oracle.com/javase/7/docs/api/java/util/regex/Matcher.html#appendReplacement(java.lang.StringBuffer,%20java.lang.String)
which provide the more clean code structure.